### ISSUE TITLE
Update GCP node provider to terminate nodes in parallel

### ISSUE THREAD

**Issue by concretevitamin** [skypilot-org/skypilot#1983]
User reported that `sky down` on multinode GCP clusters is too slow (either during failover or an actual down call). This is because the node provider impl terminates nodes serially.

We could merge in https://github.com/ray-project/ray/pull/34455 and test.

**Comment by landscapepainter** [comment]
Duplicate of #1914?

**Comment by Michaelvll** [comment]
> Duplicate of or related to #1914?

They are separate problems, as this is about the performance of `ray down`



> User reported that `sky down` on multinode GCP clusters is too slow (either during failover or an actual down call). This is because the node provider impl terminates nodes serially.
> 
> We could merge in [ray-project/ray#34455](https://github.com/ray-project/ray/pull/34455) and test.

The parallel stopping/termination in https://github.com/ray-project/ray/pull/34455 is buggy, as the gcp operation does not work well with multi-thread. Refer to https://github.com/skypilot-org/skypilot/pull/2160#issue-1783260816

This should be fixed #2199
